wasmtest/contracts-rust/contractop/src/lib.rs

```diff
@@ -0,0 +1,199 @@
+                i32.const 0  ;; storge offset
+                i32.const 20 ;; init key 0x14(KEY_MIGRATE_STORE2)
+                i32.store8   ;; store key
+                i32.const 0  ;; key addr
+                i32.const 1  ;; key size 1 byte
+                i32.const 8  ;; data addr
+                i32.const 8  ;; data is i64
+                i32.const 0  ;; always 0
+                call 1		 ;; read key to data addr of data size
+                i32.const 8  ;; should read 8
+                i32.ne
+                if
+                    unreachable ;; if read length not 8 will panic
+                end
+                i32.const 8  ;; data addr
+                i64.load
+                i64.const 206643;; data(VAL_MIGRAGE_STORE2)
+                i64.ne
+                if
+                    unreachable ;;
+                end
+                i32.const 8  ;; data addr
+                i32.const 8  ;; data is i64
+                call 0		 ;; return
+                )
+              (memory (;0;) 1)
+              (export "invoke" (func 2)))
+             **/
+            let code = &[
+                0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00, 0x01, 0x12, 0x03, 0x60, 0x00, 0x00,
+                0x60, 0x02, 0x7f, 0x7f, 0x00, 0x60, 0x05, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x01, 0x7f,
+                0x02, 0x2d, 0x02, 0x03, 0x65, 0x6e, 0x76, 0x0c, 0x6f, 0x6e, 0x74, 0x69, 0x6f, 0x5f,
+                0x72, 0x65, 0x74, 0x75, 0x72, 0x6e, 0x00, 0x01, 0x03, 0x65, 0x6e, 0x76, 0x12, 0x6f,
+                0x6e, 0x74, 0x69, 0x6f, 0x5f, 0x73, 0x74, 0x6f, 0x72, 0x61, 0x67, 0x65, 0x5f, 0x72,
+                0x65, 0x61, 0x64, 0x00, 0x02, 0x03, 0x02, 0x01, 0x00, 0x05, 0x03, 0x01, 0x00, 0x01,
+                0x07, 0x0a, 0x01, 0x06, 0x69, 0x6e, 0x76, 0x6f, 0x6b, 0x65, 0x00, 0x02, 0x0a, 0x5b,
+                0x01, 0x59, 0x00, 0x41, 0x00, 0x41, 0x25, 0x3a, 0x00, 0x00, 0x41, 0x00, 0x41, 0x01,
+                0x41, 0x08, 0x41, 0x08, 0x41, 0x00, 0x10, 0x01, 0x41, 0x08, 0x47, 0x04, 0x40, 0x00,
+                0x0b, 0x41, 0x08, 0x29, 0x03, 0x00, 0x42, 0x97, 0x85, 0xce, 0x00, 0x52, 0x04, 0x40,
+                0x00, 0x0b, 0x41, 0x00, 0x41, 0x14, 0x3a, 0x00, 0x00, 0x41, 0x00, 0x41, 0x01, 0x41,
+                0x08, 0x41, 0x08, 0x41, 0x00, 0x10, 0x01, 0x41, 0x08, 0x47, 0x04, 0x40, 0x00, 0x0b,
+                0x41, 0x08, 0x29, 0x03, 0x00, 0x42, 0xb3, 0xce, 0x0c, 0x52, 0x04, 0x40, 0x00, 0x0b,
+                0x41, 0x08, 0x41, 0x08, 0x10, 0x00, 0x0b,
+            ];
+
+            if let Some(address) =
+                runtime::contract_migrate(code, 3, "name", "version", "author", "email", "desc")
+            {
+                if let Some(resv) = runtime::call_contract(&address, &[]) {
+                    let mut source = ZeroCopySource::new(&resv);
+                    let val: u64 = source.read().unwrap();
+                    assert!(val == 0x32733);
+                } else {
+                    panic!("call migrate failed");
```

Comment:
you can use `runtime::call_contract(&address, &[]).expect("call migrate failed")`  to simplify this.

Reply:
then how to handle the val assert.